vignettes/symptom-surveys.Rmd

```diff
@@ -0,0 +1,672 @@
+
+```{r, message = FALSE, warning = FALSE}
+library(covidcast)
+library(dplyr)
+library(purrr)
+library(epipredict)
+library(recipes)
+
+case_num <- 200
+as_of_date <- "2020-05-14"
+geo_values <- covidcast_signal(
+  "jhu-csse", "confirmed_cumulative_num",
+  "2020-05-14", "2020-05-14"
+) %>%
+  filter(value >= case_num) %>%
+  pull(geo_value)
+
+# Fetch county-level Google and Facebook % CLI-in-community signals, and JHU
+# confirmed case incidence proportion
+start_day <- "2020-04-11"
+end_day <- "2020-09-01"
+
+g <- covidcast_signal("google-survey", "smoothed_cli") %>%
+  filter(geo_value %in% geo_values) %>%
+  select(geo_value, time_value, value)
+f <- covidcast_signal(
+  "fb-survey", "smoothed_hh_cmnty_cli",
+  start_day, end_day
+) %>%
+  filter(geo_value %in% geo_values) %>%
+  select(geo_value, time_value, value)
+c <- covidcast_signal(
+  "jhu-csse", "confirmed_7dav_incidence_prop",
+  start_day, end_day
+) %>%
+  filter(geo_value %in% geo_values) %>%
+  select(geo_value, time_value, value)
+
+# Rename columns
+colnames(g) <- sub("^value", "goog", colnames(g))
+colnames(f) <- sub("^value", "fb", colnames(f))
+colnames(c) <- sub("^value", "case", colnames(c))
+
+# Find "complete" counties, present in all three data signals at all times
+geo_values_complete <- intersect(
+  intersect(g$geo_value, f$geo_value),
+  c$geo_value
+)
+
+# Make one big matrix by joining these three data frames
+z <- full_join(full_join(g, f, by = c("geo_value", "time_value")),
+  c,
+  by = c("geo_value", "time_value")
+) %>%
+  filter(geo_value %in% geo_values_complete) %>%
+  as_epi_df()
+
+Logit <- function(x, a = 0.01) log((x + a) / (1 - x + a))
+Sigmd <- function(y, a = 0.01) (exp(y) * (1 + a) - a) / (1 + exp(y))
+
+#### Parameters #####
+
+# Transforms to consider, in what follows
+trans <- Logit
+inv_trans <- Sigmd
+
+# Rescale factors for our signals: bring them all down to proportions (between
+# 0 and 1)
+rescale_f <- 1e-2 # Originally a percentage
+rescale_g <- 1e-2 # Originally a percentage
+rescale_c <- 1e-5 # Originally a count per 100,000 people
+
+z <- z %>% mutate(
+  case = trans(case * rescale_c),
+  fb = trans(fb * rescale_f),
+  goog = trans(goog * rescale_g)
+)
+
+# lead = 7
+leads <- c(7, 14)
+lags <- c(0, 7, 14)
+n <- 14 # Number of trailing days to use for the training set
+
+# Nine forecast dates
+dates <- seq(as.Date("2020-05-06"), as.Date("2020-05-14"), by = "day")
+
+# List for storage of results
+out_list <- vector(mode = "list", length = length(dates))
+for (k in 1:length(dates)) {
+  date <- dates[k]
+
+  if (date %in% c("2020-05-13", "2020-05-14")) leads <- c(7, 14) else leads <- 7
+
+  # Pre-structuring test data for this date
+  z_te <- z %>%
+    rename(
+      target_date = time_value,
+      target_case = case
+    ) %>%
+    select(geo_value, target_date, target_case)
+
```

Comment:
Do the training data as well:

```r
z_tr <- z %>%
  filter(between(time_value, date - .x - max(lags) - n, date))
```

Reply:
We have the `.x` part here that depends on lead, which is why I didn't get the training data outside of `map()`.